GetUserDataProcessingActivity/handler.ts

```diff
@@ -0,0 +1,204 @@
+    kind: t.literal("QUERY_FAILURE"),
+    reason: t.string
+  }),
+  t.partial({ query: t.string })
+]);
+export type ActivityResultQueryFailure = t.TypeOf<
+  typeof ActivityResultQueryFailure
+>;
+
+export const ActivityResultFailure = t.taggedUnion("kind", [
+  ActivityResultQueryFailure,
+  ActivityResultInvalidInputFailure,
+  ActivityResultRecordNotFound
+]);
+export type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+const logPrefix = `GetUserDataProcessingActivity`;
+
+function assertNever(_: never): void {
+  throw new Error("should not have executed this");
+}
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ * @param queryName an optional name for the query, for logging purpose
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError, R>>,
+  queryName: string = ""
+) =>
+  tryCatch(lazyPromise, (err: Error) =>
+    ActivityResultQueryFailure.encode({
+      kind: "QUERY_FAILURE",
+      query: queryName,
+      reason: err.message
+    })
+  ).chain((queryErrorOrRecord: Either<QueryError, R>) =>
+    fromEither(
+      queryErrorOrRecord.mapLeft(queryError =>
+        ActivityResultQueryFailure.encode({
+          kind: "QUERY_FAILURE",
+          query: queryName,
+          reason: JSON.stringify(queryError)
+        })
+      )
+    )
+  );
+
+/**
+ * Logs depending on failure type
+ * @param context the Azure functions context
+ * @param failure the failure to log
+ */
+const logFailure = (context: Context) => (
+  failure: ActivityResultFailure
+): void => {
+  switch (failure.kind) {
+    case "INVALID_INPUT_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error decoding input|ERROR=${failure.reason}`
+      );
+      break;
+    case "QUERY_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error ${failure.query} query error |ERROR=${failure.reason}`
+      );
+      break;
+    case "RECORD_NOT_FOUND":
+      context.log.error(`${logPrefix}|Error record not found |ERROR=`);
+      break;
+    default:
+      assertNever(failure);
+  }
+};
+
+export const createGetUserDataProcessingActivityHandler = (
+  userDataProcessingModel: UserDataProcessingModel
+) => (context: Context, input: unknown): Promise<ActivityResult> => {
+  /**
+   * Updates a UserDataProcessing record by creating a new version of it with a chenged status
+   * @param param0.currentRecord the record to be modified
+   * @param param0.nextStatus: the status to assign the record to
+   *
+   * @returns either an Error or the new created record
+   */
+  const fetchRecordFromDb = ({
```

Comment:
can you extract this method outside the body of `createGetUserDataProcessingActivityHandler` ?

Reply:
naming: `fetchRecordFromDb` ? -> `getUserDataProcessingRequest`